### PR TITLE
[Merged by Bors] - chore: rename `Pi.fintype` to `Pi.instFintype`

### PR DESCRIPTION
**Pull request by b-mehta**
[Zulip discussion](https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/Pi.20instance.20names.20in.20Lean.204.20core)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

Original motivation:

> Adds instances for pi subsingleton and inhabited instances to match their other names.
This resolves a few porting notes, and should help mathport.

This is obviously moot given the time taken for this PR to land.

### CODE REVIEW

File: Mathlib/Logic/Basic.lean [hunk 1]
```diff
@@ -1288,3 +1288,10 @@ theorem dite_prop_iff_and {Q : P → Prop} {R : ¬P → Prop} [Decidable P] :
   by_cases h : P <;> simp [h, forall_prop_of_false, forall_prop_of_true]
 
 end ite
+
+-- Aliases to work around lean4#2343
+alias Pi.inhabited := instInhabitedForAll_1
+#align pi.inhabited Pi.inhabited
+
+alias Pi.subsingleton := instSubsingletonForAll
+#align pi.subsingleton Pi.subsingleton
```
Comment (eric-wieser) [inline Mathlib/Logic/Basic.lean:1297]: What reducibility does `alias` generate? I think `export` might be more appropriate than `alias` here.

**Review by eric-wieser — COMMENTED** [review]
I think we settled on `Pi.instInhabited` as the preferred name for instances, with the `inst`.

**Comment by b-mehta** [comment]
Should I also rename the other pi instances like Pi.fintype and Pi.nonempty to match this naming then?

**Comment by eric-wieser** [comment]
Can you just upstream these better names?

**Comment by b-mehta** [comment]
Scott suggested doing this instead of upstreaming the names in the linked Zulip thread, perhaps it's best to move discussion there.

File: Mathlib/Logic/Basic.lean (hunk 1 above)
Comment (b-mehta) [reply Mathlib/Logic/Basic.lean:1297]: I don't know how to use `export` to do this. Feel free to adjust.

**Comment by YaelDillies** [comment]
All the instance were separately renamed except `Pi.fintype`, so I've made this the main goal of this PR

**Pull request closed without merging**